addons/auth_signup/models/res_partner.py

```diff
@@ -63,6 +63,7 @@ def _get_signup_url_for_action(self, url=None, action=None, view_type=None, menu
             if signup_type:
                 route = 'reset_password' if signup_type == 'reset' else signup_type
 
+            query['p_id'] = partner.id
```

Comment:
Just put it directly in the definition of `query` (also probably convert it to a dict literal, there's no real reason to use a function call).